be/src/olap/data_dir.cpp

```diff
@@ -474,10 +475,12 @@ Status DataDir::load() {
         }
         if (rowset_meta->rowset_state() == RowsetStatePB::COMMITTED &&
             rowset_meta->tablet_uid() == tablet->tablet_uid()) {
-            if (!rowset_meta->get_rowset_pb().has_tablet_schema()) {
-                rowset_meta->set_tablet_schema(&tablet->tablet_schema());
+            if (!rowset_meta->tablet_schema()) {
```

Comment:
There many cases that call get_rowset_pb. For example in txn manager. If you erase schema from pb then you should add the schema to pb every time.
I think it is better to return a new pb  and add tablet schema to the new pb during get rowset pb .